misc/services/mobile-webui/mobile-webui-frontend/src/utils/index.js

```diff
@@ -15,3 +15,25 @@ export const unboxAxiosResponse = (axiosResponse) => {
     return axiosResponse.data;
   }
 };
+
+export const getLocation = (params, scanner) => {
+  const { wfProcessId, activityId, lineId, stepId, appId, altStepId } = params;
+
+  const location = `/workflow/${wfProcessId}${activityId ? `/activityId/${activityId}` : ``}${
+    lineId ? `/lineId/${lineId}` : ``
+  }${stepId ? `/stepId/${stepId}` : ``}${altStepId ? `/altStepId/${altStepId}` : ``}${
+    scanner ? `/scanner/${appId}` : ``
+  }`;
```

Comment:
this format looks a bit odd. I would split it in separate logical constants and form the location with them

Reply:
agree, imho this function shall not exist at all.
We need specific functions for each case/screen.

The fact that they are looking identical atm it's just a temporary coincidence.